proxy.go

```diff
@@ -790,45 +790,23 @@ func (p *Proxy) connect(req *http.Request) (*http.Response, net.Conn, error) {
 	}
 }
 
-func (p *Proxy) connectHTTP(req *http.Request, proxyURL *url.URL) (*http.Response, net.Conn, error) {
+func (p *Proxy) connectHTTP(req *http.Request, proxyURL *url.URL) (res *http.Response, conn net.Conn, err error) {
 	log.Debugf("martian: CONNECT with upstream HTTP proxy: %s", proxyURL.Host)
 
-	conn, err := p.dial("tcp", proxyURL.Host)
-	if err != nil {
-		return nil, nil, err
-	}
 	if proxyURL.Scheme == "https" {
-		tlsConfig := p.clientTLCConfig()
-		tlsConfig.ServerName = proxyURL.Hostname()
-		tlsConfig.NextProtos = []string{"http/1.1"}
-		conn = tls.Client(conn, tlsConfig)
-	}
-
-	pbw := bufio.NewWriter(conn)
-	pbr := bufio.NewReader(conn)
-
-	connReq := &http.Request{
-		Method: "CONNECT",
-		URL:    req.URL,
-		Host:   req.Host,
-		Header: make(http.Header),
-	}
-	if proxyURL.User != nil {
-		connReq.Header.Add("Proxy-Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(proxyURL.User.String())))
-	}
-	connReq.Write(pbw)
-
-	if err := pbw.Flush(); err != nil {
-		conn.Close()
-		return nil, nil, err
+		d := dialvia.HTTPSProxy(p.dial, proxyURL, p.clientTLCConfig())
```

Comment:
Is `clientTLCConfig()` an intentional name?

Reply:
Old typo, will fix it.